### PR TITLE
Basic understanding and usage discussion about machine learning algorithms

### PR DESCRIPTION
**Pull request by woom221**
This is an initial draft that explains the basic understanding and usage discussion about machine learning algorithms

### CODE REVIEW

**Review by hannahpan1 — CHANGES_REQUESTED** [review]

File: Topics/Machine_Learning/Understanding Machine Learning.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+Algorithms may include: k-means, Principal Component Analysis
+
+#### Reinforcement Learning
+
+Reinforement Learning involves tasks such that past actions may affect the future actions. Unlike supervised learning, it does not require predictor-target paired dataset rather, it requires a defined reward signal such that how performing a certain action could be beneficial over the others. An example would be predicting the best move for the given chess board. 
+
+Algorithms may include: Q-learning, Actor-Critic methods
+
+Key Terminologies:
```
[the remark this answers is not in the document]
Comment (hannahpan1) [reply Topics/Machine_Learning/Understanding Machine Learning.md]: Great work adding the appendix and links! Super easy to follow along!

**Review by hannahpan1 — CHANGES_REQUESTED** [review]
Some small changes to formatting will help organize your work better, see comments for more information.

File: Topics/Machine_Learning/Understanding Machine Learning.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+The first set of data is called the training dataset. As the name suggests, this dataset is used to "train" the algorithm. When we say training, we mean tuning the parameters of the algorithm to the given dataset. As this will determine the "optimal" algorithm that would solve the provided tasks, it is typical to have over 50% of the dataset as a training dataset. We may set hyperparameters to decide how the algorithm will learn.
+The second set of data serves as a verifier thus called the validation dataset. In order to avoid over-fitting, this set of data will serve as an unknown yet reusable data to verify the generalization of the fitted model. Each time a model gets trained, it will try to explain the task on the validation dataset to let the Machine Learning practitioner know its performance. While the accuracy measurement metrics may vary (i.e. one example would be a simple proportion of correct predictions), it is important to have a metric that is appropriate for the given task. Ideally, we would want to make sure the model has the minimum error on the validation set while having a relatively low error on the training set.
```
[the remark this answers is not in the document]
Comment (woom221) [reply Topics/Machine_Learning/Understanding Machine Learning.md:10]: Done! thanks for checking

**Review by hannahpan1 — APPROVED** [review]
Approved! Great job Sean, very interesting content.

**Review by hannahpan1 — APPROVED** [review]

**Pull request merged**